src/input/base-input.js

```diff
@@ -100,34 +106,54 @@ class BaseInput<T: EventTarget> extends React.Component<
   };
 
   render() {
-    const {
-      value,
-      type,
-      overrides: {
-        InputContainer: InputContainerOverride,
-        Input: InputOverride,
-        Before: BeforeOverride,
-        After: AfterOverride,
-      },
-    } = this.props;
+    const {overrides = {}, mask, maskChar, type, value} = this.props;
 
     const sharedProps = getSharedProps(this.props, this.state);
 
     const [InputContainer, inputContainerProps] = getOverrides(
-      InputContainerOverride,
+      overrides.InputContainer,
       StyledInputContainer,
     );
-    const [Input, inputProps] = getOverrides(InputOverride, StyledInput);
-    const [Before, beforeProps] = getOverrides(BeforeOverride, NullComponent);
-    const [After, afterProps] = getOverrides(AfterOverride, NullComponent);
+    const [Input, inputPropsFromOverride] = getOverrides(
+      overrides.Input,
+      StyledInput,
+    );
+    const [Before, beforeProps] = getOverrides(overrides.Before, NullComponent);
+    const [After, afterProps] = getOverrides(overrides.After, NullComponent);
+
+    if (type === CUSTOM_INPUT_TYPE.textarea) {
+      return (
+        <InputContainer {...sharedProps} {...inputContainerProps}>
+          <Before {...sharedProps} {...beforeProps} />
+          <Input
+            {...sharedProps}
+            {...this.getInputProps()}
+            {...inputPropsFromOverride}
+            value={value}
```

Comment:
The `value` is included in the `this.getInputProps`. Is there a reason we want to pass `value` after the override props?

Reply:
```suggestion
```